### PR TITLE
feat: add select all support for grids using a lazy data provider

### LINKED ISSUE #70: Select All support for Lazy Loading
It would be great to add support for select all checkbox when the lazy loading feature is activated on the grid

### PR DESCRIPTION
**Pull request by flang**
Close #70

### CODE REVIEW

File: src/main/java/com/flowingcode/vaadin/addons/gridhelpers/LazySelectAllGridHelper.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * This grid helper allows showing or hiding select all checkbox no matter if data provider is or
+ * not memory.
+ * <p>
+ * Note that enabling select all checkbox when grid uses a lazy data provider could lead to memory
+ * and performance issues, given that all the items must be eagerly loaded at the moment user clicks
+ * the checkbox.
+ * 
+ * @param <T> the type of items in grid
+ */
+public final class LazySelectAllGridHelper<T> implements Serializable {
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/gridhelpers/LazySelectAllGridHelper.java:19]: Helper classes must not be public. Any public documentation about the feature goes into `GridHelper`

**Review by javier-godoy — CHANGES_REQUESTED** [review]
See comments

File: src/main/java/com/flowingcode/vaadin/addons/gridhelpers/GridHelper.java [hunk 2]
```diff
@@ -477,6 +477,7 @@ public static double getHeightByRows(Grid<?> grid) {
    * Defines the mode in which the Grid's height is calculated.
    *
    * <p>If {@link HeightMode#CSS} is given, Grid will respect the values given via a {@code
+   * If {@link HeightMode#CSS} is given, Grid will respect the values given via a {@code
```
[the remark this answers is not in the document]
Comment (flang) [reply src/main/java/com/flowingcode/vaadin/addons/gridhelpers/GridHelper.java]: fixed

File: src/main/java/com/flowingcode/vaadin/addons/gridhelpers/LazySelectAllGridHelper.java (hunk 1 above)
Comment (flang) [reply src/main/java/com/flowingcode/vaadin/addons/gridhelpers/LazySelectAllGridHelper.java]: fixed

**Review by javier-godoy — CHANGES_REQUESTED** [review]
Please squash `fix: implement review requested changes` into `feat: add select all support for grids using a lazy data provider`, and remember to re-request review after changes are pushed.

**Review by javier-godoy — APPROVED** [review]

**Pull request merged**